app/services/analytics_events.rb

```diff
@@ -681,32 +681,224 @@ def idv_in_person_proofing_residential_address_submitted(**extra)
     track_event('IdV: in person proofing residential address submitted', **extra)
   end
 
-  def idv_in_person_proofing_address_submitted(**extra)
-    track_event('IdV: in person proofing address submitted', **extra)
+  # @param [String] flow_path
+  # @param [String] step
+  # @param [Integer] step_count
+  # @param [String] analytics_id
+  # @param [Boolean] irs_reproofing
+  # @param [Boolean] success
+  # @param [Hash] errors
+  # @param [Boolean] same_address_as_id
+  # address submitted by user
```

Comment:
Having this documented is much better than the status quo, but I recommend additionally:
- Contextualize the event description w/ the flow
- Describe the individual parameters in more detail

Reply:
I am interested in hearing more about this. In the past I've learned that comments-as-documentation can be a dangerous precedent to set. This is because they're easy to miss when updating the code and (as they can't really be tested) they are easy to miss in the future and are more likely to fall out of date as the code changes. The pattern I've learned is to focus on creating expressive names and address any nuance in the PR / commit for reference.

I can see that there are many parameters here so that reference may be both more helpful, and also quicker to fall out of date (as there are more parameters, there are more possibilities they could change).

I'm guessing this has likely been discussed before, but I'm curious if writing comments describing parameters is a best practice on the team, and when it's advised to do so.

Reply:
I feel like "comments-as-documentation" is a little bit not quite what's going on here?

These are not freeform comments, they're formatted as YARD which we then [parse, export as JSON](https://github.com/18F/identity-idp/blob/ea0b8c246f0abf1dde557033df9f801f81020e42/spec/lib/analytics_events_documenter_spec.rb#L207), and then render docs based on that ([source](https://github.com/18F/identity-handbook/blob/main/assets/js/analytics-events.tsx), [resulting docs](https://handbook.login.gov/articles/analytics-events.html))

We'll fail the build on clearly wrong kinds of comments like, param name typos. Since Ruby is a dynamic language, if a param is annotated as `String` but we pass an `Integer`, we may never know. But the main goal of this is to help enumerate and list properties that we have in our logs to make exploration and querying of them easier, which having the names and even a guess at the type does help with.

Reply:
Ohh cool, I hadn't heard of YARD before. Then you're right, my understanding was wrong. Thank you for explaining this! The examples are really helpful.

So then just for my understanding - we do this whenever we change analytics? Are there other places where we use YARD?

Reply:
I'll casually YARD document other files if I feel like it helps understand what's going on, but yes for the most part we focus/enforce on these analytics events